From 937edf3bee5f0652789f86761056b8528e178400 Mon Sep 17 00:00:00 2001
From: stefan11111 <stefan11111@shitposting.expert>
Date: Wed, 6 Aug 2025 00:07:00 +0300
Subject: [PATCH] include: export noDamageExtension

According to @felipe19930 in https://github.com/X11Libre/xserver/issues/310 ,
this symbol is required by the 470 nvidia DDX driver.

Signed-off-by: stefan11111 <stefan11111@shitposting.expert>
---
 include/extinit.h    | 5 +++++
 miext/extinit_priv.h | 1 -
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/extinit.h b/include/extinit.h
index 0c93863f78..1b584ee834 100644
--- a/include/extinit.h
+++ b/include/extinit.h
@@ -56,6 +56,11 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 /* required by: libglx */
 extern _X_EXPORT Bool noCompositeExtension;
 
+/* required by: the 470 nvidia DDX driver */
+#ifdef DAMAGE
+extern _X_EXPORT Bool noDamageExtension;
+#endif
+
 /* required by: several video drivers (eg. vmware and sis) */
 #ifdef XINERAMA
 extern _X_EXPORT Bool noPanoramiXExtension;
diff --git a/miext/extinit_priv.h b/miext/extinit_priv.h
index c892c0c6a6..4090fed7b8 100644
--- a/miext/extinit_priv.h
+++ b/miext/extinit_priv.h
@@ -8,7 +8,6 @@
 
 #include "extinit.h"
 
-extern Bool noDamageExtension;
 extern Bool noDbeExtension;
 extern Bool noDPMSExtension;
 extern Bool noGlxExtension;
